core/common/src/main/java/alluxio/collections/FieldIndex.java

```diff
@@ -61,4 +70,21 @@
    * @return the object or null if there is no such object
    */
   T getFirst(Object value);
+
+  /**
+   * Returns an iterator over the elements in this index. The elements are returned in no particular
+   * order.
+   *
+   * Note that the behavior of the iterator is unspecified if the underlying collection is modified
+   * while a thread is going through the iterator.
+   *
+   * @return an iterator over the elements in this {@link FieldIndex}
+   */
+  Iterator<T> iterator();
+
+  /**
+   * @return the number of objects in this indexed set (O(1) time for unique index, O(#fieldValue)
```

Comment:
I don't think you have to comment on the complexity of the each index implementations, since we could always add more implementations.